gesos-image-build.pipeline

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env groovy
+
+@Library(['PPCmanifest','security-ci-commons-shared-lib']) _
+
+def repoStatusReportUrl
+def gitCommitStatusContext = 'GE SOS Image Build'
+
+pipeline {
+    agent {
+        docker {
+            image 'openjdk:8'
+            label 'dind'
+        }
+    }
+    parameters {
+        string(
+            name: 'IAM_CONTAINER_CONFIG_BRANCH_NAME',
+            defaultValue: BRANCH_NAME,
+            description: 'iam-container-config repo branch name to use for GE SOS image build'
+        )
+        string(
+            name: 'IAM_CONTAINER_CONFIG_FOLDER_NAME',
+            defaultValue: 'uaa',
+            description: 'Name of container config folder with Dockerfile and prebuild script in the ' +
+                         'iam-container-config repo to use for GE SOS image build'
+        )
+        string(
+            name: 'IMAGE_TAG',
+            defaultValue: 'BUILD_TAG_FROM_BRANCH_NAME',
+            description: 'Tag for docker image to push to ECR'
+        )
+        string(
+            name: 'IMAGE_NAME',
+            defaultValue: 'iam-uaa',
+            description: 'Name for docker image to push to ECR'
+        )
+    }
+    environment {
+        GESOS_API_KEY = credentials('GESOS_API_KEY')
+        GITHUB_CREDENTIALS = credentials('GITHUB_CREDENTIALS')
+    }
+    stages {
+        stage('Github status configuration') {
+            steps {
+                script {
+                    commitSha = sh(script: "git rev-parse HEAD", returnStdout: true).trim()
+                    repoStatusReportUrl = sh(script: 'git config --get remote.origin.url', returnStdout: true).trim()
+                                                                                                              .replaceFirst('^https://([^/]+)/([^.]+)(\\.git)?$',
+                                                                                                                            'https://$1/api/v3/repos/$2/statuses/' + commitSha)
+                }
+            }
+        }
+        stage('Trigger GE SOS build pipeline') {
+            steps {
+                script {
+                    setBuildStatus(repoStatusReportUrl, gitCommitStatusContext, 'in progress', 'pending')
+                    iamContainerConfigBranchName = params.IAM_CONTAINER_CONFIG_BRANCH_NAME
+                    if (iamContainerConfigBranchName == BRANCH_NAME) {
+                        iamContainerConfigBranchName = BRANCH_NAME.replaceAll('rc_', '').replaceAll('release_', '')
+                    }
+
+                    imageTag = params.IMAGE_TAG
+                    if (imageTag == 'BUILD_TAG_FROM_BRANCH_NAME') {
+                        imageTag = iamContainerConfigBranchName.replaceAll('rc_', '').replaceAll('release_', '')
```

Comment:
According to you logic image tag is the same for rc and release.  It means you build rc - publish to ECR with 75.18.1 tag; build from release branch - overwrite 75.18.1 image in ECR...
Should we have different image tags for rc and release?

Reply:
Done, now it will push images with different tags like below

> For rc, rc_{version}
> For release, {version}